app/src/main/java/uk/ac/ebi/atlas/search/JsonGeneSearchController.java

```diff
@@ -249,7 +251,10 @@ public Set<String> getOrganismPartBySearchTerm(@RequestParam MultiValueMap<Strin
             return ImmutableSet.of();
         }
 
-        return analyticsSearchService.searchOrganismPart(geneIds.get());
+        var cellTypes = requestParams.get("cellTypes");
```

Comment:
```
if (geneIds.isEmpty()) {
            return ImmutableSet.of();
        }

        var cellTypes = requestParams.get("cellTypes");
return organismPartSearchService.search(geneIds.get(),
                cellTypes != null ? ImmutableSet.copyOf(cellTypes) : ImmutableSet.of());


```

Instead of the above, how about this code: 

```suggestion
if (!geneIds.isEmpty() && equestParams.get("cellTypes") !-null) {
     return organismPartSearchService.search(geneIds.get(), 
                ImmutableSet.copyOf(cellTypes));
 }
 else {
    return ImmutableSet.of()
 }
```

Please check the Indentation of my code in your IDE if you want to take it :). 
If you think this review is not related to endpoint improvement, ignore this.

Reply:
sorry, my bad, correct this copy and paste error - `equestParams.get("cellTypes") !-null` should be `requestParams.get("cellTypes") !=null`

Reply:
@upendrakumbham I don't really understand your comment. I don't see why is that better.
Actually that code modification would break the current logic.
Have you followed the code path that you would like me to modify?
If not, then please do. Thanks

Reply:
Sorry, my bad. I mixed up. In my opinion, Positive checks come first(! == something)->{do some work} then negative check last if(empty)-->return empty.

Reply:
Sorry, my bad. I mixed it up.